### PR TITLE
I18n/2.7 localize js

### PR DESCRIPTION
**Pull request by pglewis**
#3620

### CODE REVIEW

File: classes/PodsI18n.php [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+/**
+ * @package Pods
+ * @since 2.7
+ */
+class PodsI18n {
+
+	/**
+	 * @var PodsI18n
+	 */
+	static $instance = null;
```
Comment (sc0ttkclark) [inline classes/PodsI18n.php]: `private static`

File: classes/PodsI18n.php [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+			self::$instance = new PodsI18n();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Constructur
+	 */
+	public function __construct() {
```
Comment (sc0ttkclark) [inline classes/PodsI18n.php]: `private function`

File: classes/PodsI18n.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		 * Add strings to the localization
+		 * Setting the key of your string to the original (non translated) value is mandatory
+		 * Note: Existing keys in this class will overwrite the ones of this filter!
+		 * 
+		 * @since 2.7
+		 * @see create_strings()
+		 * @param array
+		 * @return array format: 'Untranslated string' => 'Translated string with use of WP translate functions'
+		 */
+		$strings_extra = apply_filters( 'pods_localized_strings', array() );
```
Comment (sc0ttkclark) [inline classes/PodsI18n.php]: Maybe pass `$this`? Not required though

File: classes/PodsI18n.php [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+		$this->strings = array_merge( $strings_extra, $this->strings );
```
Comment (sc0ttkclark) [inline classes/PodsI18n.php]: Should we allow people to register overrides though?

File: classes/PodsI18n.php [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+	 * We need to register them as normal string to convert to JS references
+	 * And we need to register the translations to attach to these references, these may not be variables!
+	 * 
+	 * @since 2.7
+	 */
+	public function create_strings() {
+
+		return array(
+
+				'%s is required.' =>
```
Comment (sc0ttkclark) [inline classes/PodsI18n.php]: This is interesting, I wonder if we should key these though, like 'text_is_required', 'field_is_required', 'add', 'add_new', 'add_new_record' etc

Comment (sc0ttkclark) [inline classes/PodsI18n.php]: Oh I see now, it's used by the JS, this sounds better then in that case to provide JS a way to do __( .. ) type calls

File: classes/PodsI18n.php (hunk 4 above)
Comment (JoryHogeveen) [inline classes/PodsI18n.php]: array_merge should overwrite our keys with those coming from the filter.
I'd say we want the keys added with this class to be forced as defaults.

**Pull request merged**